src/bindgen/ir/ty.rs

```diff
@@ -211,6 +211,8 @@ impl ArrayLength {
 pub enum Type {
     ConstPtr(Box<Type>),
     Ptr(Box<Type>),
+    ConstNonNullPtr(Box<Type>),
```

Comment:
Rather than this, can we have the nullability inside `ConstPtr` and `Ptr`, just like in `CDeclarator`? Seems like it'd be less duplicated code over all. Though that being said, isn't a non-nullable pointer basically just a `Ref` / `MutRef`? If so, can we use that instead?

Reply:
I've moved the nullability into `ConstPtr` and `Ptr` and gotten rid of my new variants. I didn't use `Ref`/`MutRef` because they aren't created when this code parses the syntax. I thought they might be legacy - they're only created in one place inside enumeration.rs.

Reply:
There's no good reason they can't be created during parsing.